lib/llvm-backend/src/state.rs

```diff
@@ -68,22 +69,122 @@ impl ControlFrame {
 }
 
 #[derive(Debug, Eq, PartialEq, Copy, Clone, Hash)]
-pub enum ExtraInfo {
-    None,
-
-    // This values is required to be arithmetic 32-bit NaN (or 32x4) by the WAsm
+pub struct ExtraInfo {
+    state: u8,
+}
+impl ExtraInfo {
```

Comment:
Why do we need to distinguish between `f32` and `f64` NaNs in `ExtraInfo`, since the opcodes and values are already typed?

Reply:
Because of the SIMD opcodes which aren't typed. They're all just v128, so we might go from a F64x2Add to a F32x4Mul with no cast in between.

`v128_to_f32x4` checks for a pending f64 canonicalization and applies it, while not applying a pending f32 canonicalization. `v128_to_f64x2` does the opposite.